### ISSUE TITLE
[BUG] - UVICORN_WORKERS > 1 hangs Mealie on start

### ISSUE THREAD

**Issue by ndbeals** [mealie-recipes/mealie#4354]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

When setting `UVICORN_WORKERS=2`, mealie can no longer start up.

It gets caught in an endless loop of "waiting for child process" and then "child process died" and repeat

### Steps to Reproduce

Set `UVICORN_WORKERS=2` and start mealie up.

I am using the postgres17 as my database.

### Please provide relevant logs

```
Switching to dedicated user

        User uid:    1000
        User gid:    1000

INFO     2024-10-12T14:15:46 - Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit)
INFO     2024-10-12T14:15:46 - Started parent process [1]
INFO     2024-10-12T14:15:52 - Waiting for child process [53]
INFO     2024-10-12T14:15:52 - Child process [53] died
INFO     2024-10-12T14:15:57 - Waiting for child process [54]
INFO     2024-10-12T14:15:57 - Child process [54] died
INFO     2024-10-12T14:16:03 - Waiting for child process [55]
INFO     2024-10-12T14:16:03 - Child process [55] died
INFO     2024-10-12T14:16:08 - Waiting for child process [56]
INFO     2024-10-12T14:16:08 - Child process [56] died
INFO     2024-10-12T14:16:13 - Waiting for child process [57]
```

### Mealie Version

`ghcr.io/mealie-recipes/mealie:v1.12.0`

### Deployment

Docker (Linux)

**Comment by ndbeals** [comment]
I know my setup is a little different from the default, so I tried with the default postgres docker compose from your documentation, it also has the same issue, here's the docker-compose I used:

```yaml
services:
  mealie:
    image: ghcr.io/mealie-recipes/mealie:v1.12.0
    container_name: mealie-test
    restart: always
    ports:
        - "9925:9000" # 
    deploy:
      resources:
        limits:
          memory: 1000M # 
    volumes:
      - mealie-data:/app/data/
    environment:
      # Set Backend ENV Variables Here
      ALLOW_SIGNUP: "false"
      PUID: 1000
      PGID: 1000
      TZ: America/Toronto
      UVICORN_WORKERS: 2
      MAX_WORKERS: 1
      WEB_CONCURRENCY: 1
      BASE_URL: https://mealie.yourdomain.com
      # Database Settings
      DB_ENGINE: postgres
      POSTGRES_USER: mealie
      POSTGRES_PASSWORD: mealie
      POSTGRES_SERVER: postgres
      POSTGRES_PORT: 5432
      POSTGRES_DB: mealie
    depends_on:
      postgres:
        condition: service_healthy

  postgres:
    container_name: postgres-test
    image: postgres:15
    restart: always
    volumes:
      - mealie-pgdata:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: mealie
      POSTGRES_USER: mealie
    healthcheck:
      test: ["CMD", "pg_isready"]
      interval: 30s
      timeout: 20s
      retries: 3

volumes:
  mealie-data:
  mealie-pgdata:
```

**Comment by ndbeals** [comment]
This issue is also present when using sqlite

**Comment by ndbeals** [comment]
This appears to be a bug in uvicorn, fixed by this MR: https://github.com/encode/uvicorn/pull/2373

fix was released in 0.30.2: https://github.com/encode/uvicorn/pull/2380

mealie is on unicorn v0.30.0

**Comment by michael-genson** [comment]
Thanks for the investigation! Nightly is on the [latest version of uvicorn](https://github.com/mealie-recipes/mealie/pull/4343) so it should be fixed in the next release. If you need it now you can upgrade to nightly in the meantime.
